lib/cluster.js

```diff
@@ -10,20 +10,31 @@
 
 const cluster = require('cluster');
 const Registry = require('./registry');
+const Gauge = require('./gauge');
 const util = require('./util');
 const aggregators = require('./metricAggregators').aggregators;
 
 const GET_METRICS_REQ = 'prom-client:getMetricsReq';
 const GET_METRICS_RES = 'prom-client:getMetricsRes';
+const REG_METRICS_WORKER = 'prom-client:registerMetricsWorker';
 
 let registries = [Registry.globalRegistry];
 let requestCtr = 0; // Concurrency control
 let listenersAdded = false;
+const coordinatedWorkers = new Set();
 const requests = new Map(); // Pending requests for workers' local metrics.
 
 class AggregatorRegistry extends Registry {
-	constructor() {
+	/**
+     Create an AggregatorRegistry instance. Accepts an optional `options` object.
+
+     Options are:
+        coordinated If false (default), request metrics from all cluster workers. If true, request metrics only from workers that have required prom-client.
+	 * @param {object?} options object
```

Comment:
Is the only reason this *needs* to be an option because setting it to `true` makes the order of forking vs. `new AggregatorRegistry()` matter? I'd rather go for an implementation that isn't sensitive to that, e.g. the worker repeatedly attempts to register with the master until the master acks the registration.

Having a heterogeneous pool of workers (not all of them setting up prom-client) is unusual, but I think the behavior achieved when this is `true` is what should happen by default.

Reply:
Yes, the only reason to make it an option is that it would break backwards compatibility. 

Given that a metrics client should be as unobtrusive as possible, I would be wary of adding a prolonged discovery phase. I would certainly prefer in my consuming codebase to keep things simple and accept that ordering matters. 

In the case of the homogenous cluster, the previous  implementation cleanly sidesteps a lot of the issues with “garbage collecting” the workers and nobody else had complained so far :)

Reply:
FWIW, we will be breaking backwards compatibility pretty hard soon ish (see #177, #178 and #180), so don't let semver stop you from writing the code you'd like to write :D